app/src/main/java/com/cornellappdev/uplift/ui/components/goalsetting/GoalSlider.kt

```diff
@@ -0,0 +1,152 @@
+package com.cornellappdev.uplift.ui.components.goalsetting
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.util.PRIMARY_BLACK
+import com.cornellappdev.uplift.util.PRIMARY_YELLOW
+import com.cornellappdev.uplift.util.montserratFamily
+
+/**
+ * @param value the current value of the slider
+ * @param onValueChange a callback that is called when the value of the slider changes
+ * @return composable that displays a slider for the user to select the number of days they
+ * would like to work out in a week.
+ * @sample GoalSlider(3f) {}
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun GoalSlider(
+    value: Float,
+    onValueChange: (Float) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(200.dp) // Increased to make space for labels
+            .padding(vertical = 24.dp, horizontal = 18.dp),
+        verticalArrangement = Arrangement.spacedBy(10.dp)
+    ) {
+        Text(
+            buildAnnotatedString {
```

Comment:
Woah super impressive to get this working yourself. This stuff is kinda annoying.